### PR TITLE
[refactor] add release to Makefile

### PR DESCRIPTION
**Pull request by ikaven1024**
Signed-off-by: yingjinhui <yingjinhui@didiglobal.com>

**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
Implement for https://github.com/karmada-io/karmada/pull/1814#issuecomment-1131870145

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
After run `make release`, tars are created in `_output/release` with sha256 checksum.
```
$ ls _output/release/
karmadactl-darwin-amd64.tgz         karmadactl-linux-amd64.tgz         kubectl-karmada-darwin-amd64.tgz         kubectl-karmada-linux-amd64.tgz
karmadactl-darwin-amd64.tgz.sha256  karmadactl-linux-amd64.tgz.sha256  kubectl-karmada-darwin-amd64.tgz.sha256  kubectl-karmada-linux-amd64.tgz.sha256
karmadactl-darwin-arm64.tgz         karmadactl-linux-arm64.tgz         kubectl-karmada-darwin-arm64.tgz         kubectl-karmada-linux-arm64.tgz
karmadactl-darwin-arm64.tgz.sha256  karmadactl-linux-arm64.tgz.sha256  kubectl-karmada-darwin-arm64.tgz.sha256  kubectl-karmada-linux-arm64.tgz.sha256
```

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
Thanks. Seems it's exactly we want.
/assign

I'll try it on my side later.

File: hack/release.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+release_dir="${REPO_ROOT}/_output/release"
+mkdir -p "${release_dir}"
+
+tar_file="${binary}-${os}-${arch}.tgz"
+echo "!!! Packaging ${tar_file}"
+tar czf "${release_dir}/${tar_file}" "${REPO_ROOT}/LICENSE" -C "${REPO_ROOT}/_output/bin/${os}/${arch}" "${binary}"
+
+cd "${release_dir}"
+sha256sum "${tar_file}" > "${tar_file}.sha256"
```
Comment (RainbowMango) [inline hack/release.sh:22]:
```suggestion
sha256sum "${tar_file}" > "${tar_file}.sha256-checksum.txt"
```
I'm not sure if it looks better. What do you think?

Comment (ikaven1024) [reply hack/release.sh:22]: apache also use `.sha*`

https://downloads.apache.org/tomcat/tomcat-10/v10.0.21/bin/apache-tomcat-10.0.21.zip.sha512
https://dlcdn.apache.org/httpd/httpd-2.4.53.tar.gz.sha256
https://www.apache.org/dist/kafka/1.0.0/kafka-1.0.0-src.tgz.sha512

Comment (RainbowMango) [reply hack/release.sh:22]: OK. Let's do it.
/lgtm
/approve

**Pull request merged**